src/blockmap.jl

```diff
@@ -484,6 +481,24 @@ for k in 1:8 # is 8 sufficient?
     end
 end
 
+"""
+    SparseArrays.blockdiag(As::Union{LinearMap,AbstractMatrix}...)::BlockDiagonalMap
+
+Construct a (lazy) representation of the diagonal concatenation of the arguments.
+To avoid fallback to the generic `SparseArrays.blockdiag`, there must be a `LinearMap`
+object among the first 8 arguments.    
+"""
+SparseArrays.blockdiag(As::Union{LinearMap,AbstractMatrix}...)
+
+"""
+    cat(As::Union{LinearMap,AbstractMatrix}...; dims=(1,2))::BlockDiagonalMap
+
+Construct a (lazy) representation of the diagonal concatenation of the arguments.
+To avoid fallback to the generic `Base.cat`, there must be a `LinearMap`
+object among the first 8 arguments.
+"""
+Base.cat(As::Union{LinearMap,AbstractMatrix}...; dims=dims)
```

Comment:
Is it valid to put this line of code here?

Reply:
I wasn't sure either. Would it be safe if I remove the arguments, like
```suggestion
Base.cat
```
?

Reply:
There are specific constructions to add a doc string to a function without actually defining the function. Typically you can declare a function without defining any methods as
```julia
function f end
```
without arguments. This doesn't seem to work for
```julia
function Base.cat end
```
however it does for
```julia
import Base: cat
"""
    some doc string
"""
function cat end
```

Reply:
Thanks. I have adopted that in a few more places, and it seems to work just fine.